### PR TITLE
Implement copy-to-clipboard feature

### LINKED ISSUE #3: User should be able to copy vic-machine CLI command to clipboard
**User Statement:**
As a vSphere admin I'd like to be able to copy and paste the command line string at the end of the Create VCH wizard.

**Details:**
Normally the VCH wizard will need to create a VCH by calling the vic-machine API, but there might be a case where the user simply wants to copy & paste the CLI command. This story addresses that.

**Acceptance Criteria:**

- [x] Have a select box that allows the user to select a platform for which the CLI command will be generated and copied to the clipboard

### PR DESCRIPTION
**Pull request by jooskim**
Fixes #3

### CODE REVIEW

File: ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/create-vch-wizard.service.ts [hunk 1]
```diff
@@ -81,7 +82,7 @@ export class CreateVchWizardService {
      */
     getServerGuid(): string {
         if (!this._serverGuid) {
-            // TODO: improve
+            // TODO: improve, refactor
             const userSession = <any>this.globalsService.getWebPlatform().getUserSession();
```
Comment (jak-atx) [inline ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/create-vch-wizard.service.ts]: This is a const and redefined as the same value a few times (in different functions). We should probably define it once outside of these functions.

File: ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/create-vch-wizard.service.ts [hunk 2]
```diff
@@ -81,7 +82,7 @@ export class CreateVchWizardService {
      */
```
Comment (jak-atx) [inline ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/create-vch-wizard.service.ts:80]: Remove getWipManifest() from this file if not needed

File: ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/create-vch-wizard.service.ts [hunk 3]
```diff
@@ -101,6 +102,18 @@ export class CreateVchWizardService {
     }
 
     /**
+     * Get VC's hostname
+     * TODO: unit test, refactor
+     */
+    getVcHostname(): string {
+        if (!this._serverHostname) {
+            const userSession = <any>this.globalsService.getWebPlatform().getUserSession();
+            this._serverHostname = userSession['serversInfo'][0]['name'];
```
Comment (jak-atx) [inline ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/create-vch-wizard.service.ts]: this seems a little weird. are we sure that this field will always be under position 0?

File: ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/summary/summary.component.ts [hunk 4]
```diff
@@ -27,24 +27,131 @@ export class SummaryComponent implements OnInit {
     public form: FormGroup;
     public formErrMessage = '';
     @Input() payload: any;
+    public processedPayload: any;
+    public targetOs: string;
```
Comment (jak-atx) [inline ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/summary/summary.component.ts]: what is targetOs? I see it used further below but what is it's meaning?

File: ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/summary/summary.component.ts [hunk 5]
```diff
@@ -27,24 +27,131 @@ export class SummaryComponent implements OnInit {
...
+            return;
+        }
+        this.form.get('targetOs').valueChanges
+            .subscribe(v => {
+                if (!v) {
+                    return;
+                }
+                const textareaNode = this.elementRef.nativeElement.querySelector('textarea#cliCommand');
+                this.targetOs = v;
+                this.form.get('cliCommand').setValue(this.stringifyProcessedPayload());
```
Comment (jak-atx) [inline ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/summary/summary.component.ts]: I'm not quite clear why this is part of the subscribe for targetOs

File: ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/summary/summary.html [hunk 6]
```diff
@@ -1,5 +1,5 @@
-<clr-stack-view id="vch-payload-stackview">
-  <div *ngIf="payload && payload.name && payload.computeCapacity && payload.storageCapacity && payload.networks && payload.security">
+<div *ngIf="payload && payload.name && payload.computeCapacity && payload.storageCapacity && payload.networks && payload.security">
```
Comment (jak-atx) [inline ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/summary/summary.html]: this should be broken out probably to display each one if its there. if we do it this way its all or nothing.

File: ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/summary/summary.html [hunk 7]
```diff
@@ -401,5 +401,47 @@
...
+        <div class="col-xs-6">
+          <clr-icon id="copyCliCommandToClipboardBtn" shape="clipboard" (click)="copyCliCommandToClipboard()"></clr-icon>
+        </div>
+      </div>
+      <div class="row mb10" *ngIf="copySucceeded !== null">
+        <div class="col-xs-12">
+          <div class="alert" [class.alert-success]="copySucceeded === true" [class.alert-danger]="copySucceeded === false">
+            <div class="alert-item">
+              <span class="alert-text" *ngIf="copySucceeded === true">
+                Successfully copied to clipboard
```
Comment (jak-atx) [inline ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/summary/summary.html:431]: We should validate this with UX on placement etc

File: ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/summary/summary.scss [hunk 8]
```diff
@@ -3,3 +3,12 @@ div.form-group {
     color: red;
   }
 }
+
+textarea#cliCommand {
+  min-height: 150px;
+  font-size: 0.9em;
```
Comment (jak-atx) [inline ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/summary/summary.scss:9]: Does this come from the mockup? are we overriding clarity textarea style here?

File: ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/summary/summary.scss [hunk 9]
```diff
@@ -3,3 +3,12 @@ div.form-group {
...
+}
+
+#copyCliCommandToClipboardBtn {
+  cursor: pointer;
```
Comment (jak-atx) [inline ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/summary/summary.scss]: we should probably have a generic class for this (anything that needs a pointer). or maybe clarity already provides this somewhere.

File: ui/vic-ui-h5c/vic/src/vic-webapp/src/styles.scss [hunk 10]
```diff
@@ -8,10 +8,6 @@
   left: 2% !important;
 }
 
-#vch-payload-stackview .stack-view {
-  height: 65vh !important;
```
Comment (jak-atx) [inline ui/vic-ui-h5c/vic/src/vic-webapp/src/styles.scss:12]: let's double check browser support on vh (mainly i.e, im not sure which versions we support)

File: ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/summary/summary.component.ts [hunk 11]
```diff
@@ -27,24 +27,131 @@ export class SummaryComponent implements OnInit {
...
+     * an empty array, and then return the array joined
+     * @returns {string} vic-machine compatible arguments
+     */
+    stringifyProcessedPayload(): string {
+        if (!this.targetOs) {
+            return null;
+        }
+
+        const payload = this.processedPayload;
+        const camelCasePattern = new RegExp(/([a-z])([A-Z])/g);
```
Comment (jak-atx) [inline ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/summary/summary.component.ts]: We need to stick these regexes in a util file and import them

File: ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/create-vch-wizard.service.ts (hunk 3 above)
Comment (jooskim) [reply ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/create-vch-wizard.service.ts]: will leave a TODO such that we can come back to this later for improvement

File: ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/summary/summary.component.ts (hunk 4 above)
Comment (jooskim) [reply ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/summary/summary.component.ts]: O"S"

File: ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/summary/summary.html (hunk 6 above)
Comment (jooskim) [reply ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/summary/summary.html]: maybe i should use something like payload.computeResource?.xxxxx

File: ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/summary/summary.scss (hunk 9 above)
Comment (jooskim) [reply ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/summary/summary.scss]: add a generic class in `styles.scss` for `cursor: pointer`

**Review by jak-atx — APPROVED** [review]

**Pull request merged**